src/main/sensors/battery.c

```diff
@@ -470,6 +470,16 @@ float calculateVbatPidCompensation(void) {
     return batteryScaler;
 }
 
+float calculateThrottleVbatLimitFactor(void)
+{
+    float factor =  1.0f;
+    if (batteryConfig()->voltageMeterSource != VOLTAGE_METER_NONE && batteryCellCount > 0) {
+        float vbat = (float) voltageMeter.filtered / batteryCellCount;
+        factor = constrainf((float) batteryConfig()->vbatwarningcellvoltage / vbat, 0.0f, 1.0f);
```

Comment:
What range do you expect to get for `factor`?

Reply:
Factor can vary from 0 to 1.

The constrainf has been used to reduce variables pollution:

it could have been written as follows:

factor = vwarn / max(vbat, vwarn);

I can change it if it becomes more readeable

Reply:
So what setting for `batteryConfig()->vbatwarningcellvoltage` and what type of battery would you use to get a factor of 0?

Reply:
As math tells, 0 will never be obtained

Reply:
Please answer questions that you are asked on pull requests that you are opening.

Reply:
Sorry, which other question?

Here the answer is, never.

For vbat < vwarn, the factor is 1, otherwise it would be vwarn / vbat, that can never be zero.

A zero factor would just make the throttle be 0.

Reply:
@tylercorleone:

> Here the answer is, never.

Ok, this establishes that your previous answer 

> Factor can vary from 0 to 1.

was simply wrong. Can I get you to answer the original question with something that isn't incorrect then.

Reply:
Math says its range is (0, 1]. I expect it to vary from 0.8 to 1. It depends on battery warning voltage and the current voltage.